test/language/comments/hashbang-function-constructor.js

```diff
@@ -0,0 +1,23 @@
+/*---
+esid: pending
+description: >
+    Hashbang comments should not be allowed in function evaluator contexts.
+info: |
+    HashbangComment::
+      #! SingleLineCommentChars[opt]
+flags: [module]
+---*/
+const AsyncFunction = (async function (){}).constructor;
+const GeneratorFunction = (function *(){}).constructor;
+const AsyncGeneratorFunction = (async function *(){}).constructor;
+for (ctor of [
+  Function,
+  AsyncFunction,
+  GeneratorFunction,
+  AsyncGeneratorFunction,
+]) {
+  assert.throws(SyntaxError, () => ctor('#!\n_',''));
+  assert.throws(SyntaxError, () => ctor('#!\n_'));
+  assert.throws(SyntaxError, () => new ctor('#!\n_',''));
+  assert.throws(SyntaxError, () => new ctor('#!\n_'));
```

Comment:
```js
  assert.throws(SyntaxError, () => ctor('#!\n_',''), `${ctor.name} #1`);
  assert.throws(SyntaxError, () => ctor('#!\n_'), `${ctor.name} #2`);
  assert.throws(SyntaxError, () => new ctor('#!\n_',''), `${ctor.name} #3`);
  assert.throws(SyntaxError, () => new ctor('#!\n_'), `${ctor.name} #4`);
```

WDYT? this helps mapping the assertions to whoever is consuming this test file

Reply:
can we also add assertions for hashbangs placed at the first part of a function block?

```js
function fn() {#!
}
```